core/controller/src/main/scala/whisk/core/loadBalancer/DistributedLoadBalancerData.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.core.loadBalancer
+
+import akka.actor.ActorSystem
+import akka.util.Timeout
+import akka.pattern.ask
+import whisk.common.Logging
+import whisk.core.entity.{ActivationId, UUID}
+
+import scala.collection.concurrent.TrieMap
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
+/**
+ * Encapsulates data used for loadbalancer and active-ack bookkeeping.
+ *
+ * Note: The state keeping is backed by distributed akka actors. All CRUDs operations are done on local values, thus
+ * a stale value might be read.
+ */
+class DistributedLoadBalancerData(seedNodesProvider: SeedNodesProvider,
+                                  implicit val actorSystem: ActorSystem,
+                                  logging: Logging)
+    extends LoadBalancerData {
+
+  implicit val timeout = Timeout(5.seconds)
+  implicit val executionContext = actorSystem.dispatcher
+  private val activationsById = TrieMap[ActivationId, ActivationEntry]()
+
+  private val seedNodes = {
+    val nodes = seedNodesProvider.getSeedNodes()
+    if (nodes.size > 0)
+      nodes
+    else {
+      logging.error(
+        this,
+        "Either seed nodes were not provided or not specified in the correct " +
+          "format, e.g. whitespace " +
+          "separated <IP:PORT>")
+      sys.exit(1)
```

Comment:
Can we avoid exiting anywhere in the code?
Can we instead let the creation of `DistributedLoadBalancerData` let fail, which lets the creation of the Loadbalancer fail?